applications/CoSimulationApplication/custom_utilities/conversion_utilities.h

```diff
@@ -59,10 +60,77 @@ class KRATOS_API(CO_SIMULATION_APPLICATION) ConversionUtilities
-        const Variable<array_1d<double,3> >& rNodalVariable );
+        const Variable<TDataType>& rElementalVariable,
+        const Variable<TDataType>& rNodalVariable );
+
+
+    /**
+     * @brief Converts elemental data to nodal data directly. Nodal value is a sum of elemental values divided by the number of elements.
+     * 
+     * This function takes elemental data from a specified variable and converts it to nodal data,
+     * storing the result in another specified variable. The conversion is performed directly
+     * without any intermediate steps.
+     * 
+     * @tparam TDataType The type of the data to be converted.
+     * @param rModelPart The model part containing the elements and nodes.
+     * @param rElementalVariable The variable containing the elemental data to be converted.
+     * @param rNodalVariable The variable where the converted nodal data will be stored.
+     */
+    template<class TDataType>
+    static void ConvertElementalDataToNodalDataDirect(
+        ModelPart& rModelPart,
+        const Variable<TDataType>& rElementalVariable,
+        const Variable<TDataType>& rNodalVariable );
+
+
+    /**
+     * @brief Converts nodal data to elemental data. Element value is a sum of nodal values divided by the number of nodes.
+     *
+     * This utility function transfers data from nodal variables to elemental variables
+     * within a given ModelPart. It is templated to work with various data types.
+     *
+     * @tparam TDataType The type of data to be converted.
+     * @param rModelPart The ModelPart containing the nodes and elements.
+     * @param rElementalVariable The variable to store the converted elemental data.
+     * @param rNodalVariable The variable containing the nodal data to be converted.
+     */
+    template<class TDataType>
+    static void ConvertNodalDataToElementalDataDirect(
+        ModelPart& rModelPart,
+        const Variable<TDataType>& rElementalVariable,
+        const Variable<TDataType>& rNodalVariable );
+
+        /**
+     * @brief Converts nodal data to elemental data using transpose operator. Element value is a sum of nodal value contribution, which is split equally to all neighbouring elements.
+     *
+     * This utility function transfers data from nodal variables to elemental variables
+     * within a given ModelPart. It is templated to work with various data types.
+     *
+     * @tparam TDataType The type of data to be converted.
+     * @param rModelPart The ModelPart containing the nodes and elements.
+     * @param rElementalVariable The variable to store the converted elemental data.
+     * @param rNodalVariable The variable containing the nodal data to be converted.
+     */
+    template<class TDataType>
+    static void ConvertNodalDataToElementalDataTranspose(
+        ModelPart& rModelPart,
+        const Variable<TDataType>& rElementalVariable,
+        const Variable<TDataType>& rNodalVariable );
```

Comment:
You need to explain what you mean by the transpose operator here. My first though would be transposing whatever value you're converting, but that's not at all what's happening.

Reply:
I wrote in the description of the function what it does. Should I improve the description?

Reply:
I understand the description, but have no idea how it relates to a transpose operator. When I read transpose, I think of a column vector turning into a row vector, or matrix transposition, etc, since we're talking about variables and not mapping.

Reply:
"Besides, I really dislike the consistent/conservative terminology bcs its strongly tied to FSI, which CoSim is not
I use "distribute_values" in many places for this purpose (or "use_transpose" for the mapper)"

Reply:
I don't see how that ties in here. As I see it, these functions define a mapping between a piecewise constant field (over element geometries) and a bi/trilinear field (or whatever shape functions you have).

Maybe I'm just out of my depth here and @philbucher can help me with terminology. In any case, I' change the `Transpose` naming to something less confusing.

Reply:
I prefer consistent/conservative as it keeps the sum of the field values or keeps the positioning of the value.